--- a/webpack.config.js	2017-11-03 19:41:59.847729765 +0100
+++ b/webpack.config.js	2017-11-03 19:41:00.897727305 +0100
@@ -1,5 +1,6 @@
 const fs = require('fs');
 const path = require('path');
+const webpack = require('webpack');
 const CopyWebpackPlugin = require('copy-webpack-plugin');
 const ProgressPlugin = require('webpack/lib/ProgressPlugin');
 const CircularDependencyPlugin = require('circular-dependency-plugin');
@@ -536,6 +537,9 @@
       "main"
     ]
   },
+  "externals": [
+    resolveExternals
+  ],
   "resolveLoader": {
     "modules": [
       "./node_modules",
@@ -917,6 +921,9 @@
       "tsConfigPath": "src/tsconfig.app.json",
       "skipCodeGeneration": true,
       "compilerOptions": {}
+    }),
+    new webpack.ProvidePlugin({
+      __extends: 'typescript-extends'
     })
   ],
   "node": {
@@ -928,9 +935,25 @@
     "process": true,
     "module": false,
     "clearImmediate": false,
-    "setImmediate": false
+    "setImmediate": false,
+    "__dirname": true
   },
   "devServer": {
     "historyApiFallback": true
   }
 };
+
+function resolveExternals(context, request, callback) {
+  return resolveMeteor(request, callback) ||
+    callback();
+}
+
+function resolveMeteor(request, callback) {
+  var match = request.match(/^meteor\/(.+)$/);
+  var pack = match && match[1];
+
+  if (pack) {
+    callback(null, 'Package["' + pack + '"]');
+    return true;
+  }
+}
